### DISCUSSION TITLE
Bumps of the same dependency version are grouped together

### DISCUSSION THREAD

**Discussion by gkranis** [renovatebot/renovate#19213] [Suggest an Idea]
It looks like bumps of the same dependency version are grouped together. Even without any grouping option enabled.

One example of why this might be undesirable is the following

Imagine an ansible scenario.
inventory/production/group_vars/foo.yml:
``` yaml
mydep:
  image: docker.io/mydep:FOO
```
inventory/test/group_vars/foo.yml:
``` yaml
mydep:
  image: docker.io/mydep:FOO
```

Currently renovate will just bump both references in the same PR. I would prefer if it created separate PRs, so that I can test the changes in the `test` environment first before merging the PR that affects the `production` environment

**Comment by gkranis** [comment]
I think for this case I can define 2 groups with different `matchFiles` and it should work how want

**Reply by gkranis** [reply]
the `matchFiles` trick works, but it means that for each dependency i will have to create one rule per deployment environment...

**Reply by viceice** [reply]
please read the docs more closely. use match paths with additional branch prefix. then you should max need one rule per repo root folder

**Reply by gkranis** [reply]
Here's my current 'trick'
```
  "packageRules": [
    {
      "matchManagers": ["ansible"],
      "matchPackageNames": ["docker.io/mydep"],
      "matchPaths": ["inventory/production/"],
      "groupName": "mydep-production"
    },
    {
      "matchManagers": ["ansible"],
      "matchPackageNames": ["docker.io/mydep"],
      "matchPaths": ["inventory/test/"],
      "groupName": "mydep-test"
    },
    {
      "matchManagers": ["ansible"],
      "matchPackageNames": ["docker.io/mydep2"],
      "matchPaths": ["inventory/production/"],
      "groupName": "mydep2-production"
    },
    {
      "matchManagers": ["ansible"],
      "matchPackageNames": ["docker.io/mydep2"],
      "matchPaths": ["inventory/test/"],
      "groupName": "mydep2-test"
    },
]
```

Heres what you are suggesting i guess
```
 "packageRules": [
    {
      "matchPaths": ["inventory/production/"],
      "additionalBranchPrefix": "production/"
    },
    {
      "matchPaths": ["inventory/test/"],
      "additionalBranchPrefix": "test/"
    }
]
```

thanks, will try it

**Reply by rarkins** [reply]
If you have too many directories, `additionalBranchPrefix` can use templated too, such as `{{packageFileDir}}` or `{{parentDir}}`. Search this repo for examples and check out https://docs.renovatebot.com/templates/
